xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -463,3 +483,105 @@ func circuitBreakersFromCluster(cluster *v3clusterpb.Cluster) *uint32 {
 	}
 	return nil
 }
+
+// outlierConfigFromCluster extracts the relevant outlier detection configuration from
+// the received cluster resource. Returns nil if no OutlierDetection.
+func outlierConfigFromCluster(cluster *v3clusterpb.Cluster) (*OutlierDetection, error) {
+	// Corresponds to this line from A50: "If the outlier_detection field is not
+	// set in the Cluster message, a "no-op" outlier_detection config will be
+	// generated"
```

Comment:
You could skip this comment in favor of having a function level comment which talks about what values are returned when.

Reply:
Deleted this comment. The function level comment now states "outlierConfigFromCluster extracts the relevant outlier detection configuration from the received cluster resource. Returns nil if no OutlierDetection field set in the cluster resource."